### ISSUE TITLE
Add the ability to reverse complement sequences before writing them

### ISSUE THREAD

**Issue by ElDeveloper** [qiime2/q2-quality-filter#33]
`split_libraries_fastq.py` provided had a `--rev_comp` flag, that would let you reverse complement the sequences before they were written out.

After talking with @wasade, we both agree that such functionality might not be appropriate for `q2-quality-filter`, but we couldn't really figure out where this would fit better, so also opening this issue hoping this can get relocated to the proper repo.

**Comment by jairideout** [comment]
q2-demux?

**Comment by ElDeveloper** [comment]
Maybe, though then there's no way to reverse complement per-sample FASTQ files.

**Comment by jairideout** [comment]
Are you wanting support for reverse complementing any kind of sequence data stored in `.qza` files, not just during demultiplexing?

**Comment by ElDeveloper** [comment]
Yes. This was the case in QIIME1, since the functionality was available in `split_libraries_fastq.py` and `multiple_split_libraries_fastq.py` used `split_libraries_fastq.py`.

**Comment by jairideout** [comment]
Sounds good! Are you available to work on this for the 2017.9 release?

**Comment by ElDeveloper** [comment]
Yes, where should this be added to?

**Comment by gregcaporaso** [comment]
What's the use case for reverse complementing fastq files? I worry that that might lead to assumptions about the quality profiles being violated (e.g., median quality will be positively correlated with sequence position in the resulting files, rather than negatively correlated, which could mess things up downstream). It seems like a reverse complement action might make more sense if it operated on `FeatureData[Sequence]` instead.

**Comment by ElDeveloper** [comment]
@gregcaporaso, after solving the problem I had with that particular dataset, I don't think I have a use-case anymore. Seems like the only place where this could matter would be with the `FeatureData[Sequence]` as you mention. I am down to put that in place if you all think this still makes sense.

**Comment by ebolyen** [comment]
Since it looks like this isn't immediately necessary, we're dropping it from our sprint plan. We'll keep an eye out for more situations where this could be needed.

**Comment by ElDeveloper** [comment]
Sounds good to me.

**Comment by ARW-UBT** [comment]
Would the following be a use case?
We continously analyse fungal ITS amplicons (paired-end data). Due to the length heterogeneitiy of ITS, it is not possible to joine forward and reverse reads for many of them. Therefore, joined reads cannot be processed by 
a) q2 quality-filter q-score-joined (followed by q2 deblur denoise-other) or by 
b) q2 dada2 denoise-paired. 
I such situations, one can process single end data from R1 reads (forward), but this fails for R2 reads, probably because the 'antisense' R2 sequences are sorted out at q2 deblur denoise-other. 

Since independent analyses can be combined in q2, I thought about 
1) reverse-complement the FeatureData[sequence] artifact
2) analyse the rc-R2-sequences as single-end data and
3) finally combine both analysis results...

Best regards
